File: src/API/WesternStatesWater.WestDaat.Contracts.Client/PolygonEtDataCollection.cs
﻿namespace WesternStatesWater.WestDaat.Contracts.Client;

public class PolygonEtDataCollection
{
    public Guid? WaterConservationApplicationEstimateLocationId { get; set; }

    public string PolygonWkt { get; set; }

    public Common.DataContracts.DrawToolType DrawToolType { get; set; }

    public double AverageYearlyTotalEtInInches { get; set; }

    public double AverageYearlyTotalEtInAcreFeet { get; set; }

    public double? AverageYearlyNetEtInInches { get; set; }

    public double? AverageYearlyNetEtInAcreFeet { get; set; }

    public GeometryEtDatapoint[] Datapoints { get; set; }
}

File: src/DashboardUI/src/data-contracts/PolygonEtDataCollection.ts
import { GeometryEtDatapoint } from './GeometryEtDatapoint';

export interface PolygonEtDataCollection {
  waterConservationApplicationEstimateLocationId: string | null;
  polygonWkt: string;
  averageYearlyTotalEtInInches: number;
  averageYearlyTotalEtInAcreFeet: number;
  averageYearlyNetEtInInches: number | null;
  averageYearlyNetEtInAcreFeet: number | null;
  datapoints: GeometryEtDatapoint[];
}
